node/src/accountant/mod.rs

```diff
@@ -1579,22 +1660,189 @@ mod tests {
+            .search_for_indispensable_adjustment_result(Ok(Either::Right(AdjustmentAnalysis::new(
+                Adjustment::ByServiceFee,
+                vec![make_analyzed_account(123)],
+            ))))
+            .adjust_payments_result(Err(PaymentAdjusterError::AllAccountsEliminated));
+
+        test_handling_payment_adjuster_error(test_name, payment_adjuster);
+
+        let log_handler = TestLogHandler::new();
+        log_handler.exists_log_containing(&format!(
+            "WARN: {test_name}: Payment adjustment has not produced any executable payments. Please \
+        add funds into your consuming wallet in order to avoid bans from your creditors. Details: \
+        The adjustment algorithm had to eliminate each payable from the recently urged payment due \
+        to lack of resources"
+        ));
+        log_handler
+            .exists_log_containing(&format!("INFO: {test_name}: The Payables scan ended in"));
+        log_handler.exists_log_containing(&format!(
+            "ERROR: {test_name}: Payable scanner could not finish. If matured payables stay untreated \
+            long, your creditors may impose a ban on you"
+        ));
+    }
+
+    #[test]
+    fn payment_adjuster_error_is_not_reported_to_ui_if_scan_not_manually_requested() {
+        init_test_logging();
+        let test_name =
+            "payment_adjuster_error_is_not_reported_to_ui_if_scan_not_manually_requested";
+        let mut subject = AccountantBuilder::default().build();
+        let payment_adjuster = PaymentAdjusterMock::default()
+            .search_for_indispensable_adjustment_result(Err(
+                PaymentAdjusterError::EarlyNotEnoughFeeForSingleTransaction {
+                    number_of_accounts: 20,
+                    transaction_fee_opt: Some(TransactionFeeImmoderateInsufficiency {
+                        per_transaction_requirement_minor: 40_000_000_000,
+                        cw_transaction_fee_balance_minor: U256::from(123),
+                    }),
+                    service_fee_opt: None,
+                },
+            ));
+        let payable_scanner = PayableScannerBuilder::new()
+            .payment_adjuster(payment_adjuster)
+            .build();
+        subject.logger = Logger::new(test_name);
+        subject.scanners.payable = Box::new(payable_scanner);
+        let qualified_payable = make_non_guaranteed_qualified_payable(111_111);
+        let protected_payables = protect_qualified_payables_in_test(vec![qualified_payable]);
+        let blockchain_agent = BlockchainAgentMock::default();
+        let msg = BlockchainAgentWithContextMessage::new(
+            protected_payables,
+            Box::new(blockchain_agent),
+            None,
+        );
+
+        subject.handle_payable_payment_setup(msg);
+
+        // Test didn't blow up while the subject was unbound to other actors
+        // therefore we didn't attempt to send the NodeUiMessage
```

Comment:
Let's make this comment say:

```
// No NodeUiMessage was sent because there's no `response_skeleton`, it is evident by the fact that the test didn't blow up even though UIGateway is unbound
```